### DISCUSSION TITLE
Enhanced CNN Support

### DISCUSSION THREAD

**Discussion by pulsipher** [cog-imperial/OMLT#68] [Ideas]
Currently, 2D convolutional layers can be read into OMLT. To better support CNNs, the following layers should also be accommodated:
- Pooling layers (i.e., the ones summarized [here](https://keras.io/api/layers/pooling_layers/))
- 1D convolutional layers (maybe 3D?)

These additions would enable the use of typical 1D and 2D CNNs (maybe 3D if desired...).

**Comment by zshiqiang** [comment]
Many thanks for these ideas! We raise an issue about enhancing CNN in OMLT, please see [#129](https://github.com/cog-imperial/OMLT/issues/129).
